src/pages/index.tsx

```diff
@@ -16,6 +16,25 @@ export default function Home() {
       refetchOnWindowFocus: false,
     }
   );
+  const addDeckMutation = trpc.useMutation('add-deck');
+
+  const [decks, setDecks] = React.useState<Deck[]>(data?.decks || []);
+
+  const handleDeckSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const name = (e as any).target.elements.name;
+
+    const input = {
+      name: name.value,
+      type: "default",
+      userId: userQuery.data?.user?.id
+    };
+    const deck = await addDeckMutation.mutateAsync(input);
+    decks.push(deck);
+    setDecks([...decks]);
```

Comment:
If you wanted to be trickier, I think you could change these two lines to one (according to the documentation):
`setDecks(prevDeck => prevDeck.push(deck))`

Not that I think that's necessary--just something kinda fancy from the documentation: https://reactjs.org/docs/hooks-reference.html#usestate